### DISCUSSION TITLE
Welcome to NeqSim discussions

### DISCUSSION THREAD

**Discussion by EvenSol** [equinor/neqsim#193] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

You find documentation on the [NeqSim homepage](https://equinor.github.io/neqsimhome/), the [Java Wiki](https://github.com/equinor/neqsim/wiki), the [Python Wiki](https://github.com/equinor/neqsimpython/wiki) and the [Matlab Wiki](https://github.com/equinor/neqsimmatlab/wiki).

**Comment by GrandZangoule** [comment]
Hello all, I have a couple of questions:
1. How do I get the mole fraction of each component in the different phases?
2. Does the package calculate Dew/Bubble point Temperatures and pressures?

**Reply by EvenSol** [reply]
Hi.
1.
How do I get the mole fraction of each component in the different phases?

#Mole fraction of a component in total fluid
fluid1.getComponent('methane').getz()

#Get mole fraction of a phase (moles in phase/total number of moles)
fluid1.getPhase('gas').getMoleFraction()

#Mole fraction of component in a phase of type gas, oil or aqueous:
fluid1.getPhase('gas').getComponent('methane').getx()
fluid1.getPhase('oil').getComponent('methane').getx()
fluid1.getPhase('aqueous').getComponent('methane').getx()

#Mole fraction of component in phase number 0..n - where order is from lightest to heaviest:
fluid1.getPhase(0).getComponent('methane').getx()

#and mole fraction for first added component( component number 0) in th lightest phase:
fluid1.getPhase(0).getComponent(0).getx()

#molar composition
fluid1.getMolarComposition()

You can see some examples here:
https://colab.research.google.com/github/EvenSol/NeqSim-Colab/blob/master/notebooks/emptytemplate.ipynb


2. Does the package calculate Dew/Bubble point Temperatures and pressures?
Yes. You use the dew point temperature and dew point pressure methods in https://github.com/equinor/neqsim/blob/master/src/main/java/neqsim/thermodynamicOperations/ThermodynamicOperations.java

**Reply by GrandZangoule** [reply]
Thank you very much for your quick response. Your answer to question #1 was exactly what I needed.
For question #2, I could not find the function to use to calculate or obtain the Bubble or Dew point. I tried a few functions but had errors. Please help with the correct function calls.

Thanks again.

**Reply by EvenSol** [reply]
These examples might be usefull:
https://github.com/EvenSol/NeqSim-Colab/blob/master/notebooks/gasquality/hydrocarbon_dew_point_of_natural_gas.ipynb.

You can run the following script in Python:


```

# Calculation of dew and bubble points

import neqsim
from neqsim.thermo.thermoTools import dewt, bubt,bubp, dewp, fluid
import matplotlib.pyplot as plt

fluid0_HC = fluid('srk')
fluid0_HC.addComponent('methane',0.7)
fluid0_HC.addComponent('ethane',0.1)
fluid0_HC.addComponent('propane',0.1)
fluid0_HC.addComponent('n-butane',0.1)
fluid0_HC.setMixingRule("classic")

fluid0_HC.setPressure(10.0, 'bara')
fluid0_HC.setTemperature(0.0, 'C')

dewtemperature = dewt(fluid0_HC)
print('dew point temperature ', fluid0_HC.getTemperature('K'), ' K', ' pressure ', fluid0_HC.getPressure('bara'), ' bara')

#set inital guess
fluid0_HC.setTemperature(-140.0, 'C')
bubtemperature = bubt(fluid0_HC)
print('bubble point temperature ', fluid0_HC.getTemperature('K'), ' K', fluid0_HC.getPressure('bara'), ' bara')

#set inital guess
fluid0_HC.setTemperature(-50.0, 'C')
fluid0_HC.setPressure(30.0, 'bara')
bubpressure =bubp(fluid0_HC)
print('bubble point pressure ', fluid0_HC.getPressure('bara'), ' bara ',  fluid0_HC.getTemperature('K'), ' K')

#set inital guess
fluid0_HC.setTemperature(275.0, 'K')
fluid0_HC.setPressure(30.0, 'bara')
dewpressure =dewp(fluid0_HC)
print('dew point pressure ', fluid0_HC.getPressure('bara'), ' bara ',  fluid0_HC.getTemperature('K'), ' K')

from neqsim.neqsimpython import jNeqSim
thermoOps = jNeqSim.thermodynamicOperations.ThermodynamicOperations(fluid0_HC)
thermoOps.calcPTphaseEnvelope()

plt.plot(list(thermoOps.getOperation().get("dewT")),list(thermoOps.getOperation().get("dewP")), label="dew point")
plt.plot(list(thermoOps.getOperation().get("bubT")),list(thermoOps.getOperation().get("bubP")), label="bubble point")
plt.title('PT envelope')
plt.xlabel('Temperature [K]')
plt.ylabel('Pressure [bar]')
plt.legend()
plt.show()
```
Output:

dew point temperature  274.8507677589821  K  pressure  10.0  bara
bubble point temperature  155.42940315451594  K 10.0  bara
bubble point pressure  65.1502714869753  bara  223.14999999999998  K
dew point pressure  10.06055839453803  bara  275.0  K

![image](https://github.com/equinor/neqsim/assets/41290109/1bb12887-bdf8-4213-89da-d5351c108103)

**Reply by GrandZangoule** [reply]
Worked perfectly. Thank you very much. Your routines are awesome. I have some ideas I'm working on and will like to collaborate with your team on the near future.

Thanks again.

**Comment by GrandZangoule** [comment]
How do I suppress this and any other messages? 
**error could not calculate**

**Reply by EvenSol** [reply]
The message is printed in the thermoTools.py that you import and use. One option can be to call neqsim directly and not via thermoTools. 

The other option will be to disable logging (however this is not available in the current release).
import logging
logging.disable(logging.CRITICAL)

You can test it from github: 
pip install git+https://github.com/equinor/neqsim-python.git

**Comment by UmudMomun** [comment]
Hello everyone,

I am a beginner in Google Earth Engine (GEE) and I want to detect methane plumes using satellite data. However, I am having trouble implementing this in GEE because my coding and remote sensing background is not strong.

Could anyone share some example scripts, tutorials, or step-by-step guidance on how to identify methane plumes in GEE? Any help or resources would be greatly appreciated!

Thank you very much!

Best regards,  
VM

**Reply by EvenSol** [reply]
Dear VM.
This is a very good suggestion. We are working with google earth engine in relation to NeqSim and field development and will put out some examples in future. I would be glad to hear how you end up getting methane plums satellite data into google earth.

When it comes to methane emsissions and neqsim you can have look at the following papers:
1. [Industrial Case Study: Application of Current Capabilities in Methane Measurement at Equinor's Hammerfest LNG Facility](https://www.ilmexhibitions.com/methane/abstract/Industrial+Case+Study%3A+Application+of+Current+Capabilities+in+Methane+Measurement+at+Equinor%27s+Hammerfest+LNG+Facility/2214/), 2025
2. Virtual Measurement of Emissions from Produced Water Using an Online Process Simulator , TEKNA produced water conference, 2025
